user/src/com/google/gwt/user/server/rpc/impl/ServerSerializationStreamWriter.java

```diff
@@ -422,30 +321,14 @@ private static String escapeString(String toEscape, boolean splitNodes,
     int i = 0;
     while (i < length) {
 
-      // Add one segment at a time, up to maxNodeLength characters. Note this always leave room
-      // for at least 6 characters at the end (maximum unicode escaped character size).
-      int maxSegmentVectorSize = splitNodes
-          ? (charVector.getSize() + MAX_STRING_NODE_LENGTH - 5)
-          : Integer.MAX_VALUE;
-
-      while (i < length && charVector.getSize() < maxSegmentVectorSize) {
+      while (i < length) {
```

Comment:
This can be removed. The same loop definition is in the line above.

Reply:
Removed the extra loop